tests/forms_tests/tests/test_forms.py

```diff
@@ -3398,6 +3398,48 @@ class CommentForm(Form):
 <div class="errorlist"><div class="error">This field is required.</div></div>
 <p>Comment: <input type="text" name="comment" required /></p>""")
 
+    def test_html_error_escaping(self):
+        class TestForm(Form):
+            foo = CharField(widget=HiddenInput(), required=False)
+            bar = CharField()
+
+            def clean_foo(self):
+                raise ValidationError("Foo & 'bar'!")
+
+            def clean_bar(self):
+                raise ValidationError("Foo & 'bar'!")
+
+        form = TestForm({
+            'foo': "bar",
+            'bar': 'foo'
+        })
+        form.is_valid()
+
+        self.assertHTMLEqual(
+            form.as_ul(),
+            '<li><ul class="errorlist nonfield"><li>(Hidden field foo) Foo &amp; &#39;bar&#39;!</li></ul></li>'
+            '<li><ul class="errorlist"><li>Foo &amp; &#39;bar&#39;!</li></ul><label for="id_bar">Bar:</label> '
+            '<input type="text" name="bar" value="foo" required id="id_bar" />'
+            '<input type="hidden" name="foo" value="bar" id="id_foo" /></li>'
+        )
+
+        self.assertHTMLEqual(
+            form.as_ul(),
```

Comment:
This is the same as the previous assertion. I'd think only 1 assertionis needed since all the `as_*` methods use `_html_output()`.